### PR TITLE
syscall Stat fixes

### PR DESCRIPTION
**Pull request by dgryski**

### CODE REVIEW

**Comment by dgryski** [comment]
`archive/tar` passes on Darwin and would otherwise pass on wasi but `internal/testenv` doesn't know that wasi doesn't have symlinks (and we don't support `t.Skipf`).

File: src/syscall/syscall_libc_darwin.go [hunk 1]
```diff
@@ -168,10 +173,10 @@ type Stat_t struct {
 	Gid       uint32
 	Rdev      int32
 	Pad_cgo_0 [4]byte
-	Atim      Timespec
-	Mtim      Timespec
-	Ctim      Timespec
-	Btim      Timespec
+	Atimespec Timespec
```
Comment (dkegel-fastly) [inline src/syscall/syscall_libc_darwin.go:175]: Probably want to remove the misleading comment above, mea culpa.

**Comment by dkegel-fastly** [comment]
If that last commit allows archive/tar to build and pass, should it also add it to TEST_PACKAGES_DARWIN or TEST_PACKAGES_LINUX in the toplevel Makefile?

File: src/syscall/syscall_libc_darwin.go (hunk 1 above)
Comment (dgryski) [reply src/syscall/syscall_libc_darwin.go:175]: Good point. I'll remove the comment.

**Comment by dgryski** [comment]
Yeah was thinking about adding it for Darwin. Need the see what's preventing it from running on Linux.

**Comment by dgryski** [comment]
For the record,
```
~/go/src/go.googlesource.com/go/src/archive/tar $ egrep '(go:build)|([AC]tim)' stat_actime*.go
stat_actime1.go://go:build aix || linux || dragonfly || openbsd || solaris
stat_actime1.go:func statAtime(st *syscall.Stat_t) time.Time {
stat_actime1.go:	return time.Unix(st.Atim.Unix())
stat_actime1.go:func statCtime(st *syscall.Stat_t) time.Time {
stat_actime1.go:	return time.Unix(st.Ctim.Unix())
stat_actime2.go://go:build darwin || freebsd || netbsd
stat_actime2.go:func statAtime(st *syscall.Stat_t) time.Time {
stat_actime2.go:	return time.Unix(st.Atimespec.Unix())
stat_actime2.go:func statCtime(st *syscall.Stat_t) time.Time {
stat_actime2.go:	return time.Unix(st.Ctimespec.Unix())
```

**Comment by dgryski** [comment]
Looks like this probably needs `-tags=osusergo`, but adding that to all the testing make rules seems like a larger change than I'd like for a PR about fixing the `Stat_t` field names.

**Review by dkegel-fastly — APPROVED** [review]
LGTM

Would have liked this commit to include some change to the tests that gives some evidence the change improves things, but if that's hard, fine.

**Pull request merged**